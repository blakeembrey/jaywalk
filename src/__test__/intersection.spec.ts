import { validationFunction } from '../'

describe('Intersection', () => {
  it('should work with intersection types', () => {
    const validate = validationFunction({
      '@type': 'Intersection',
      type: [
        {
          '@type': 'String',
          minLength: 5
        },
        {
          '@type': 'String',
          maxLength: 10
        }
      ]
    })

    const input = 'testing'

    return validate(input).then(function (output) {
      expect(output).toEqual(input)
    })
  })

  it('should reject invalid intersection values', () => {
    const validate = validationFunction({
      '@type': 'Intersection',
      type: [
        {
          '@type': 'String',
          minLength: 5
        },
        {
          '@type': 'String',
          maxLength: 10
        }
      ]
    })

    expect.assertions(1)

    return validate('hey').catch((err) => {
      expect(err.errors).toMatchSnapshot()
    })
  })
})
