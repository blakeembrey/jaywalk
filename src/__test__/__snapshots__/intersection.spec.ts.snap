// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Intersection should reject invalid intersection values 1`] = `
Array [
  Object {
    "dataPath": Array [],
    "message": "min length",
    "schemaPath": Array [
      "type",
      0,
      "minLength",
    ],
  },
]
`;
